Client/GuzzleApiClient.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Bridge for the Content API.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\BridgeBundle\Client;
+
+use Superdesk\ContentApiSdk\API\Request\RequestInterface;
+use Superdesk\ContentApiSdk\API\Request\OAuthDecorator;
+use Superdesk\ContentApiSdk\API\Response;
+use Superdesk\ContentApiSdk\ContentApiSdk;
+use Superdesk\ContentApiSdk\Client\AbstractApiClient;
+use Superdesk\ContentApiSdk\Client\ClientInterface;
+use Superdesk\ContentApiSdk\Exception\AccessDeniedException;
+use Superdesk\ContentApiSdk\Exception\AuthenticationException;
+use Superdesk\ContentApiSdk\Exception\ClientException;
+use Superdesk\ContentApiSdk\Exception\ContentApiException;
+use Superdesk\ContentApiSdk\Exception\ResponseException;
+
+/**
+ * Request service that implements all method regarding basic request/response
+ * handling.
+ */
+class GuzzleApiClient extends AbstractApiClient
+{
+    /**
+     * Default request headers.
+     *
+     * @var array
+     */
+    protected $headers = array(
+        'Accept' => 'application/json'
+    );
+
+    /**
+     * Options which come from Bundle configuration.
+     *
+     * @var array
+     */
+    protected $options = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function makeApiCall(RequestInterface $request)
+    {
+        $response = null;
+
+        if ($this->authenticator->getAccessToken() !== null) {
```

Comment:
I would move this if statement to private method instead. The code will be more readable and the method wont be so long